tests/certification/state/ravendb/ravendb_test.go

```diff
@@ -0,0 +1,255 @@
+		return stateStore
+	}, "ravenDb")
+
+	currentGrpcPort := ports[0]
+	currentHTTPPort := ports[1]
+
+	basicTest := func(ctx flow.Context) error {
+		client, err := daprClient.NewClientWithPort(strconv.Itoa(currentGrpcPort))
+		if err != nil {
+			panic(err)
+		}
+		defer client.Close()
+
+		err = client.SaveState(ctx, stateStoreName, certificationTestPrefix+"key1", []byte("ravenCert1"), nil)
+		require.NoError(t, err)
+
+		err = client.SaveState(ctx, stateStoreName, certificationTestPrefix+"key2", []byte("ravenCert2"), nil)
+		require.NoError(t, err)
+
+		// get state
+		item, err := client.GetState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		require.NoError(t, err)
+		assert.Equal(t, "ravenCert1", string(item.Value))
+
+		errUpdate := client.SaveState(ctx, stateStoreName, certificationTestPrefix+"key1", []byte("ravenCertUpdate"), nil)
+		require.NoError(t, errUpdate)
+		item, errUpdatedGet := client.GetState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		require.NoError(t, errUpdatedGet)
+		assert.Equal(t, "ravenCertUpdate", string(item.Value))
+
+		// delete state
+		err = client.DeleteState(ctx, stateStoreName, certificationTestPrefix+"key1", nil)
+		require.NoError(t, err)
+
+		return nil
+	}
+
+	eTagTest := func() func(ctx flow.Context) error {
+		return func(ctx flow.Context) error {
+			client, err := daprClient.NewClientWithPort(strconv.Itoa(currentGrpcPort))
+			if err != nil {
+				panic(err)
+			}
+			defer client.Close()
+
+			err = client.SaveState(ctx, stateStoreName, "k", []byte("v1"), nil)
+			require.NoError(t, err)
+
+			resp1, err := client.GetState(ctx, stateStoreName, "k", nil)
+			require.NoError(t, err)
+
+			err = client.SaveStateWithETag(ctx, stateStoreName, "k", []byte("v2"), resp1.Etag, nil)
+			require.NoError(t, err)
+
+			resp2, err := client.GetState(ctx, stateStoreName, "k", nil)
+			require.NoError(t, err)
+
+			err = client.SaveStateWithETag(ctx, stateStoreName, "k", []byte("v3"), "900invalid", nil)
+			require.Error(t, err)
+
+			resp3, err := client.GetState(ctx, stateStoreName, "k", nil)
+			require.NoError(t, err)
+			assert.Equal(t, resp2.Etag, resp3.Etag)
+			assert.Equal(t, "v2", string(resp3.Value))
+
+			return nil
+		}
+	}
+
+	timeToLiveTest := func() func(ctx flow.Context) error {
+		return func(ctx flow.Context) error {
+			client, err := daprClient.NewClientWithPort(fmt.Sprint(currentGrpcPort))
+			require.NoError(t, err)
+			defer client.Close()
+
+			assert.Error(t, client.SaveState(ctx, stateStoreName, certificationTestPrefix+"ttl1", []byte("revendbCert"), map[string]string{
+				"ttlInSeconds": "mock value",
+			}))
+			require.NoError(t, client.SaveState(ctx, stateStoreName, certificationTestPrefix+"ttl2", []byte("revendbCert2"), map[string]string{
+				"ttlInSeconds": "-1",
+			}))
+			require.NoError(t, client.SaveState(ctx, stateStoreName, certificationTestPrefix+"ttl3", []byte("revendbCert3"), map[string]string{
+				"ttlInSeconds": "3",
```

Comment:
I suggest we make this a higher value cause CI runners are slow and sometimes the key can expire before we get it.